cpu/kinetis/periph/i2c.c

```diff
@@ -45,474 +51,515 @@
 #endif
 
 /**
- * @brief Array holding one pre-initialized mutex for each I2C device
+ * @brief   Thread flag used internally for signaling between ISR and user thread
  */
-static mutex_t locks[] =  {
-#if I2C_0_EN
-    [I2C_0] = MUTEX_INIT,
-#endif
-#if I2C_1_EN
-    [I2C_1] = MUTEX_INIT,
-#endif
-#if I2C_2_EN
-    [I2C_2] = MUTEX_INIT
-#endif
-#if I2C_3_EN
-    [I2C_3] = MUTEX_INIT
-#endif
+#define THREAD_FLAG_KINETIS_I2C (1u << 8)
```

Comment:
Added this to avoid dependency on contested PR #9284. We can update the driver later if a consensus is reached regarding the reserved thread flags